sdk/src/utils/public/price-math.ts

```diff
@@ -107,9 +107,18 @@ export class PriceMath {
   }
 
   public static priceToTickIndex(price: Decimal, decimalsA: number, decimalsB: number): number {
-    return PriceMath.sqrtPriceX64ToTickIndex(
-      PriceMath.priceToSqrtPriceX64(price, decimalsA, decimalsB)
-    );
+    const sqrtPriceX64 = PriceMath.priceToSqrtPriceX64(price, decimalsA, decimalsB);
+    if (sqrtPriceX64.gt(new BN(MAX_SQRT_PRICE))) {
+      console.warn("Provided price is too high, returning max tick index.");
+      return MAX_TICK_INDEX;
+    }
+
+    if (sqrtPriceX64.lt(new BN(MIN_SQRT_PRICE))) {
+      console.warn("Provided price is too low, returning min tick index.");
+      return MIN_TICK_INDEX;
+    }
+
+    return PriceMath.sqrtPriceX64ToTickIndex(sqrtPriceX64);
   }
 
   public static priceToInitializableTickIndex(
```

Comment:
if the caller specified a price that was too low or too high, `sqrtPriceX64ToTickIndex` would throw an exception. if the price is too low or too high, return the min or max valid tick index